docs/_includes/generated/expert_high_availability_zk_section.html

```diff
@@ -14,6 +14,12 @@
             <td>String</td>
             <td>Defines the ACL (open|creator) to be configured on ZK node. The configuration value can be set to “creator” if the ZooKeeper server configuration has the “authProvider” property mapped to use SASLAuthenticationProvider and the cluster is configured to run in secure mode (Kerberos).</td>
         </tr>
+        <tr>
+            <td><h5>high-availability.zookeeper.client.connection-loss-tolerate</h5></td>
+            <td style="word-wrap: break-word;">true</td>
+            <td>Boolean</td>
+            <td>Defines whether or not leader election tolerates connection loss exception. Enable this option will improve the stability when ZK ensemble in an unstable network environment. For technical details, see FLINK-10052.</td>
```

Comment:
How long is the connection loss allowed to be? Does it wait forever, or is there some timeout?

Why do we allow disabling this feature?

Reply:
Curator beyond 3.0 will send a session expire event when it believes the expire timeout elapsed. So briefly, it doesn't wait forever because it will receive a  `ConnectionState.LOST` event and the `LeaderLatch` revokes leadership and re-contends. Anyway, it is a configuration in Curator scope.

Yes I'd prefer always turn on this feature. React to `ConnectionState.SUSPENDED` often turns out to be the cause of instability. If there is anything we gain from disable this feature, it will be

- Give up leadership fast, so that we don't request modification on JobGraphStore(etc.) which will be later retried by Curator and successfully modify without leadership. But even we disable it we don't solve the problem. See also FLINK-10333. The general case is, user suffers from unstable zk connection while leadership doesn't switch.

Reply:
Is this curator timeout configurable somehow?

Reply:
It is an existing option 

https://github.com/apache/flink/blob/2a4a2e22c65756d95d3c378d42be11bb583bc01d/flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java#L131

And here is the tech details if you're interested in it.

https://github.com/apache/curator/blob/959c1ca34f6ebcb11370180dfafbd8e85320dcd2/curator-framework/src/main/java/org/apache/curator/framework/state/ConnectionState.java#L66-L81